es6-lib/upstream/core.js

```diff
@@ -108,31 +71,36 @@ class Core {
       if (err) return onComplete(err);
       this._log('DeleteDataset');
       request.del({
-          url: `${url}/views/${layer.uid}`,
-          headers: this._headers(),
-          json: true
-        })
+        url: `${url}/views/${layer.uid}`,
+        headers: this._headers(),
+        json: true
+      })
         .on('response', this._onResponseStart(onComplete))
         .on('error', this._onErrorResponse(onComplete));
     });
   }
 
-  create(layer, onComplete) {
-    if(layer.uid !== Layer.EMPTY) {
+  create(parentUid, layer, onComplete) {
+    if (layer.uid !== Layer.EMPTY) {
       logger.warn(`Layer uid is not empty, layer uid is ${layer.uid}, cannot create layer in datastore!`);
     }
     return this._url((err, url) => {
       if (err) return onComplete(err);
 
       this._log(`CreateDataset request to core`);
       request.post({
-          url: `${url}/views?nbe=true`,
-          headers: this._headers(),
-          body: {
-            name: layer.name
-          },
-          json: true
-        })
+        url: `${url}/views?nbe=true`,
+        headers: this._headers(),
+        body: {
+          name: layer.name,
+          displayType: 'geoRows',
+          privateMetadata: {
+            isNbe: true,
+            parentUid
```

Comment:
Not sure I understand how this works. It doesn't need to be in dictionary like format?